File: bruteforce/b2503/Solution.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.Arrays;

public class Solution {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws NumberFormatException, IOException {
        int n = Integer.parseInt(br.readLine());
        ArrayList<int[]> hints = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            hints.add(Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray());
        }
        br.close();

        int result = 0;

        for (int x = 123; x <= 987; x++) {
            String strX = String.valueOf(x);

            if (strX.charAt(0) == '0' || strX.charAt(1) == '0' || strX.charAt(2) == '0') {
                continue;
            }
            if (strX.charAt(0) == strX.charAt(1)
                    || strX.charAt(1) == strX.charAt(2)
                    || strX.charAt(2) == strX.charAt(0)) {
                continue;
            }

            int passCnt = 0;
            for (int[] hint : hints) {
                int num = hint[0];
                int strike = hint[1];
                int ball = hint[2];

                String strNum = String.valueOf(num);
                int strikeCnt = 0;
                int ballCnt = 0;
                for (int i = 0; i < 3; i++) {
                    if (strX.contains(String.valueOf(strNum.charAt(i)))) {
                        if (strX.charAt(i) == strNum.charAt(i)) {
                            strikeCnt++;
                        } else {
                            ballCnt++;
                        }
                    }
                }
                if (strikeCnt == strike && ballCnt == ball) {
                    passCnt++;
                }
            }

            if (passCnt == n) {
                result++;
            }
        }

        bw.write(result + "\n");
        bw.flush();
        bw.close();
    }

}

File: bruteforce/b2503/solution.py
import sys

input = sys.stdin.readline

n = int(input())
hints = [tuple(map(int, input().split())) for _ in range(n)]

result = 0
for x in range(123, 988):
    str_x = str(x)
    if str_x[0] == '0' or str_x[1] == '0' or str_x[2] == '0':
        continue
    if str_x[0] == str_x[1] or str_x[1] == str_x[2] or str_x[2] == str_x[0]:
        continue

    pass_cnt = 0
    for number, s, b in hints:
        s_cnt = 0
        b_cnt = 0
        for i in range(3):
            if str(number)[i] in str_x:
                if str_x[i] == str(number)[i]:
                    s_cnt += 1
                else:
                    b_cnt += 1

        if s_cnt == s and b_cnt == b:
            pass_cnt += 1

    if pass_cnt == n:
        result += 1

print(result)
